src/Operation/CreateSearchIndexes.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/*
+ * Copyright 2015-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB\Operation;
+
+use MongoDB\Driver\Command;
+use MongoDB\Driver\Exception\RuntimeException as DriverRuntimeException;
+use MongoDB\Driver\Server;
+use MongoDB\Exception\InvalidArgumentException;
+use MongoDB\Exception\UnsupportedException;
+use MongoDB\Model\SearchIndexInput;
+
+use function array_column;
+use function array_is_list;
+use function current;
+use function MongoDB\is_document;
+use function sprintf;
+
+/**
+ * Operation for the createIndexes command.
+ *
+ * @see \MongoDB\Collection::createSearchIndex()
+ * @see \MongoDB\Collection::createSearchIndexes()
+ * @see https://mongodb.com/docs/manual/reference/command/createSearchIndexes/
+ */
+class CreateSearchIndexes implements Executable
+{
+    private string $databaseName;
+    private string $collectionName;
+    private array $indexes = [];
+    private array $options;
+
+    /**
+     * Constructs a createSearchIndexes command.
+     *
+     * @param string                 $databaseName   Database name
+     * @param string                 $collectionName Collection name
+     * @param list<array|object>     $indexes        List of search index specifications
+     * @param array{comment?: mixed} $options        Command options
+     * @throws InvalidArgumentException for parameter parsing errors
+     */
+    public function __construct(string $databaseName, string $collectionName, array $indexes, array $options)
+    {
+        if (! array_is_list($indexes)) {
+            throw new InvalidArgumentException('$indexes is not a list');
+        }
+
+        foreach ($indexes as $i => $index) {
+            if (! is_document($index)) {
+                throw InvalidArgumentException::expectedDocumentType(sprintf('$indexes[%d]', $i), $index);
+            }
+
+            $this->indexes[] = new SearchIndexInput((array) $index);
```

Comment:
This is documented as accepting documents for `$index`, but you perform an array cast here. What happens if a user passes a document in a form other than array or stdClass here (e.g. `MongoDB\BSON\Document` or a Serializable instance)?

For this reason, I think it'd make sense to require `$indexes` be a list of arrays. That would be consistent to how CreateIndexes works today (it uses `! is_array($index)` for validation). That behavior is also covered by a `testConstructorRequiresIndexSpecificationsToBeAnArray()` unit test.

Reply:
I added the restriction to accept only arrays.